app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCTableView.scala

```diff
@@ -191,15 +191,21 @@ class DLCTableView(model: DLCPaneModel) {
               }
             }
           }
-          // Enable/disable menu items per selection
+          // Enable/disable menu items per selection state
           row.item.onChange { (_, _, newContent) =>
             if (newContent != null) {
-              cancelDLCItem.disable =
-                row.item.value.isInstanceOf[AcceptedDLCStatus]
-              val enableRefundExecute =
-                row.item.value.isInstanceOf[BroadcastedDLCStatus]
-              refundDLCItem.disable = !enableRefundExecute
-              executeDLCItem.disable = !enableRefundExecute
+              cancelDLCItem.disable = row.item.value.state match {
+                case DLCState.Offered | DLCState.Accepted | DLCState.Signed =>
+                  false
+                case _ => true
```

Comment:
we tend to want to avoid doing `case _ =>` as if we add a new state we won't have the compiler yell at us that we need to handle it. Should explicity list all of the possible states

Reply:
This is a known subset we want to select for, so I have a bit of a visceral reaction to naming out every single thing we know we don't want when _ is setup to be that and the in-group assumed to be a known into the future and guaranteeing that this code would require updated now for underlying type updates.

I understand why there's reason to be explicit to every case in underlying wallet cases, but at the UI, this obfuscates intent without much to gain.

I checked in a full explicit class callout.

Reply:
I guess another option would be to use double equals notation on an explicit subset if the goal is to not use match with a _. I can change the code to that notation if it's more desirable.

Reply:
to simplify something you could do is add a `canClose` and a `canCancel` function to the states. Then can just do 
```scala
cancelDLCItem.disable = row.item.value.state.canCancel
```